rideshare.rb

```diff
@@ -0,0 +1,211 @@
+
+
+trip_array = [
+  {
+    driver_id: "DR0004",
+    date: "3rd Feb 2016",
+    cost: 5,
+    rider_id: "RD0022",
+    rating: 5,
+  },
+  {
+    driver_id: "DR0001",
+    date: "3rd Feb 2016",
+    cost: 10,
+    rider_id: "RD0003",
+    rating: 3,
+  },
+  {
+    driver_id: "DR0002",
+    date: "3rd Feb 2016",
+    cost: 25,
+    rider_id: "RD0073",
+    rating: 5,
+  },
+  {
+    driver_id: "DR0001",
+    date: "3rd Feb 2016",
+    cost: 30,
+    rider_id: "RD0015",
+    rating: 4,
+  },
+  {
+    driver_id: "DR0003",
+    date: "4th Feb 2016",
+    cost: 5,
+    rider_id: "RD0066",
+    rating: 5,
+  },
+  {
+    driver_id: "DR0004",
+    date: "4th Feb 2016",
+    cost: 10,
+    rider_id: "RD0022",
+    rating: 4,
+  },
+  {
+    driver_id: "DR0002",
+    date: "4th Feb 2016",
+    cost: 15,
+    rider_id: "RD0013",
+    rating: 1,
+  },
+  {
+    driver_id: "DR0003",
+    date: "5th Feb 2016",
+    cost: 50,
+    rider_id: "RD0003",
+    rating: 2,
+  },
+  {
+    driver_id: "DR0002",
+    date: "5th Feb 2016",
+    cost: 35,
+    rider_id: "RD0066",
+    rating: 3,
+  },
+  {
+    driver_id: "DR0004",
+    date: "5th Feb 2016",
+    cost: 20,
+    rider_id: "RD0073",
+    rating: 5,
+  },
+  {
+    driver_id: "DR0001",
+    date: "5th Feb 2016",
+    cost: 45,
+    rider_id: "RD0003",
+    rating: 2,
+  },
+]
+
+########################################################
+# Step 4: Total Driver's Earnings and Number of Rides
+
+def print_hashes_nicely(array_of_hashes)
+  array_of_hashes.each do |item|
+    puts "\t#{item.keys[0]}: #{item.values[0]}"
+  end
```

Comment:
I love the way you've broken out all this functionality into separate methods. Good work!